includes/Blocks/Block.php

```diff
@@ -122,65 +123,184 @@ private function register_block_attributes_as_fields(): void {
 	}
 
 	/**
-	 * Gets the WPGraphQL field registration config for the block attributes.
-	 *
-	 * @param ?array $block_attributes The block attributes.
+	 * Returns the type of the block attribute
+	 * 
+	 * @param string $name The block name
+	 * @param object $attribute The block attribute config
+	 * @param string $prefix Current prefix string to use for the get_query_type
 	 */
-	private function get_block_attribute_fields( ?array $block_attributes ): array {
-		$block_attribute_fields = [];
-
-		// Bail early if no attributes are defined.
-		if ( null === $block_attributes ) {
-			return $block_attribute_fields;
-		}
-
-		foreach ( $block_attributes as $attribute_name => $attribute_config ) {
-			$graphql_type = null;
-
-			if ( ! isset( $attribute_config['type'] ) ) {
-				return $block_attribute_fields;
-			}
+	private function get_attribute_type( $name, $attribute, $prefix ): mixed {
+		$type = null;
 
-			switch ( $attribute_config['type'] ) {
+		if ( isset( $attribute['type'] ) ) {
+			switch ( $attribute['type'] ) {
 				case 'string':
-					$graphql_type = 'String';
+					$type = 'String';
+					break;
+				case 'boolean':
+					$type = 'Boolean';
 					break;
 				case 'number':
-					$graphql_type = 'Float';
+					$type = 'Float';
 					break;
 				case 'integer':
-					$graphql_type = 'Int';
-					break;
-				case 'boolean':
-					$graphql_type = 'Boolean';
+					$type = 'Int';
 					break;
 				case 'array':
+					if ( isset( $attribute['query'] ) ) {
+						$type = [ 'list_of' => $this->get_query_type( $name, $attribute['query'], $prefix ) ];
+					} elseif ( isset( $attribute['items'] ) ) {
+						$of_type = $this->get_attribute_type( $name, $attribute['items'], $prefix );
+
+						if ( null !== $of_type ) {
+							$type = [ 'list_of' => $of_type ];
+						} else {
+							$type = Scalar::get_block_attributes_array_type_name();
+						}
+					} else {
+						$type = Scalar::get_block_attributes_array_type_name();
+					}
+					break;
 				case 'object':
-					$graphql_type = Scalar::get_block_attributes_object_type_name();
+					$type = Scalar::get_block_attributes_object_type_name();
 					break;
 			}
+		} elseif ( isset( $attribute['source'] ) ) {
+			$type = 'String';
+		}
+
+		if ( null !== $type ) {
+			$default_value = $attribute['default'] ?? null;
+
+			if ( isset( $default_value ) ) {
+				$type = [ 'non_null' => $type ];
+			}
+		}
+
+		return $type;
+	}
+
+	/**
+	 * Gets the WPGraphQL field registration config for the block attributes.
+	 *
+	 * @param ?array $block_attributes The block attributes.
+	 * @param string $prefix The current prefix string to use for the get_query_type
+	 * 
+	 * @return array
+	 */
+	private function get_block_attribute_fields( ?array $block_attributes, $prefix = null ): array {
```

Comment:
`$prefix = null` doesn't match the phpdoc type of `string`. I'd recommend `$prefix = ''` here to match.